### PR TITLE
*: move EndpointSelection doc to godoc

### PR DESCRIPTION
**Pull request by gyuho**
This merges two redundant documentation into one.

As Jon pointed out, this suits better in godoc (https://github.com/coreos/etcd/pull/4231#issuecomment-173571107).

### CODE REVIEW

File: Documentation/go_client.md [hunk 1]
```diff
@@ -1,16 +0,0 @@
-# etcd go client API
-
-This document describes using the etcd client Go package `github.com/coreos/etcd/client`.
-
-## Endpoint Selection Mode
-
-Programs call `client.New()` to create an etcd client instance, passing configuration through the `client.Config` parameter. `client.Config`'s `SelectionMode` member is an `EndpointSelectionMode` `enum` that specifies the policy for choosing the etcd cluster node to which requests are sent.
```
Comment (jonboulle) [inline Documentation/go_client.md]: I would add something along the lines of this bit to the actual member godoc
> "`client.Config`'s `SelectionMode` member is an `EndpointSelectionMode` `enum` that specifies the policy for choosing the etcd cluster node to which requests are sent."

**Comment by jonboulle** [comment]
thanks for doing this!

File: client/client.go [hunk 2]
```diff
@@ -119,7 +129,8 @@ type Config struct {
 	// A HeaderTimeoutPerRequest of zero means no timeout.
 	HeaderTimeoutPerRequest time.Duration
 
-	// SelectionMode specifies a way of selecting destination endpoint.
+	// SelectionMode is an EndpointSelectionMode enum that specifies the
+	// policy for choosing the etcd cluster node to which requests are sent.
```
Comment (gyuho) [inline client/client.go]: @jonboulle Just added this line. PTAL. Thanks!

**Comment by jonboulle** [comment]
Lgtm

**Comment by gyuho** [comment]
Thanks, will merge after CI passes.

**Comment by mitake** [comment]
@gyuho @jonboulle thanks for doing this. godoc seems to be a suitable place for the doc.

**Pull request merged**